packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -588,6 +611,72 @@ export class NftDetectionController extends BaseController<
               ? elm.blockaidResult?.result_type === BlockaidResultType.Benign
               : true),
         );
+        // Retrieve collections from apiNfts
+        const collections = apiNfts.reduce<string[]>((acc, currValue) => {
+          if (!acc.includes(currValue.token.contract)) {
+            acc.push(currValue.token.contract);
+          }
+          return acc;
+        }, []);
+
+        // Call API to retrive collections infos
+        // The api accept a max of 20 contracts
+        const collectionResponse: GetCollectionsResponse =
+          await reduceInBatchesSerially({
+            values: collections,
+            batchSize: MAX_GET_COLLECTION_BATCH_SIZE,
+            eachBatch: async (allResponses, batch) => {
+              const params = new URLSearchParams(
+                batch.map((s) => ['contract', s]),
+              );
+              params.append('chainId', '1'); // Adding chainId 1 because we are only detecting for mainnet
+              const collectionResponseForBatch = await fetchWithErrorHandling({
+                url: `${
+                  NFT_API_BASE_URL as string
+                }/collections?${params.toString()}`,
+                options: {
+                  headers: {
+                    Version: NFT_API_VERSION,
+                  },
+                },
+                timeout: NFT_API_TIMEOUT,
+              });
+
+              return {
+                ...allResponses,
+                ...collectionResponseForBatch,
+              };
+            },
+            initialResult: {},
+          });
+
+        // Add collections response fields to  newnfts
+        if (collectionResponse.collections?.length) {
+          apiNfts.forEach((singleNFT) => {
+            const found = collectionResponse.collections.find(
+              (elm) =>
+                elm.id?.toLowerCase() ===
+                singleNFT.token.contract.toLowerCase(),
+            );
+            if (found) {
+              singleNFT.token = {
+                ...singleNFT.token,
+                collection: {
+                  ...(singleNFT.token.collection
+                    ? singleNFT.token.collection
+                    : {}),
+                  creator: found?.creator,
+                  openseaVerificationStatus: found?.openseaVerificationStatus,
+                  contractDeployedAt: found.contractDeployedAt,
+                  ownerCount: found.ownerCount,
+                  tokenCount: found.tokenCount,
+                  topBid: found.topBid,
```

Comment:
Same as my other comment but wasn't clear why the collection call was needed for most of these.  Everything except creator and ownerCount seems to already be in the /users responses?

Reply:
Added response in comment above ☝️  🙏